src/proof/lfsc/lfsc_node_converter.h

```diff
@@ -133,6 +133,13 @@ class LfscNodeConverter : public NodeConverter
                                            const std::string& name);
   /** Should we traverse n? */
   bool shouldTraverse(Node n) override;
+  /**
+   * Make APPLY_UF, which ensures the operator op is a variable. If it is not,
+   * we create a dummy variable whose name is the result of printing op. This
+   * is to ensure proper smt2 printing, which does not permit operators to
+   * be higher-order terms.
+   */
+  Node mkApplyUf(Node op, const std::vector<Node>& args) const;
```

Comment:
this will be useful for the Lean printer as well. Maybe add to the node converter?

Reply:
I think we should make an intermediate class to inherit from if we do this, since I'd prefer keeping the interface to NodeConverter relatively minimal for the sake of clarity.

Reply:
Sounds good. Let's wait then until I start merging the Lean stuff.